src/iuse.cpp

```diff
@@ -2488,6 +2488,36 @@ int iuse::makemound( player *p, item *it, bool t, const tripoint & )
     }
 }
 
+int iuse::clear_snow( player *p, item *it, bool t, const tripoint & )
+{
+    if( !p || t ) {
+        return 0;
+    }
+
+    const cata::optional<tripoint> pnt_ = choose_adjacent( _( "Clear snow where?" ) );
+    if( !pnt_ ) {
+        return 0;
+    }
+    tripoint pnt = *pnt_;
+
+    if( pnt == p->pos() ) {
+        add_msg( m_info, _( "You clear your sinuses." ) );
+        return 0;
+    }
+    int moves;
+    if( g->m.furn( pnt ) == f_snow ) {
+        moves = MINUTES( 120 ) / it->get_quality( DIG ) * 100;
+    } else {
+        p->add_msg_if_player( _( "There isn't any snow there." ) );
+        return 0;
+    }
```

Comment:
This kind of code is a bit dangerous. `moves` is mutable and can become uninitialized if someone decides to remove the `return` statement. 
```suggestion

    if( g->m.furn( pnt ) != f_snow ) {
        p->add_msg_if_player( _( "There isn't any snow there." ) );
        return 0;
    }
    const int moves = MINUTES( 120 ) / it->get_quality( DIG ) * 100;    
```